### PR TITLE
Add Microsoft.SqlServer.Server documentation

### PR DESCRIPTION
**Pull request by DavoudEshtehari**
Introducing the new hierarchy to support Microsoft.SqlServer.Server documentation.

### CODE REVIEW

File: doc/snippets/Microsoft.SqlServer.Server/InvalidUdtException.xml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        <AssemblyName>Microsoft.SqlServer.Server</AssemblyName>
+        <AssemblyVersion>1.0.0.0</AssemblyVersion>
+      </AssemblyInfo>
+      <ReturnValue>
+        <ReturnType>Microsoft.SqlServer.Server.InvalidUdtException</ReturnType>
+      </ReturnValue>
+      <Parameters>
+        <Parameter Name="udtType" Type="System.Type" Index="0" FrameworkAlternate="netcore-2.1;netcore-2.2;netcore-3.0;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0" />
+        <Parameter Name="resourceReason" Type="System.String" Index="1" FrameworkAlternate="netcore-2.1;netcore-2.2;netcore-3.0;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0" />
+      </Parameters>
```
Comment (DavoudEshtehari) [inline doc/snippets/Microsoft.SqlServer.Server/InvalidUdtException.xml:74]: I have no idea about the `Member signature` tag except for C# here!

**Review by DavoudEshtehari — COMMENTED** [review]
@David-Engel  What do you think about the XML format originally from SDS?

File: doc/snippets/Microsoft.SqlServer.Server/InvalidUdtException.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        <Create>
+            <param name="udtType">The <see cref="T:System.Type" /> object.</param>
+            <param name="resourceReason">The <see cref="T:System.String" /> object that represents a string in string resources. The default value is `SqlUdtReason_NoUdtAttribute` which looks up a localized string similar to "no UDT attribute".</param>
+            <summary>Creates a new <see cref="T:Microsoft.SqlServer.Server.InvalidUdtException" /> object.</summary>
+            <returns>A new <see cref="T:Microsoft.SqlServer.Server.InvalidUdtException" /> object.</returns>
+            <remarks>
+            <format type="text/markdown"><![CDATA[  
+  
+## Remarks  
+This method Looks up a localized string similar to "&apos;{0}&apos; is an invalid user defined type, reason: {1}.", and fills it in order with `udtType` and `resourceReason` resolved value.
```
Comment (David-Engel) [inline doc/snippets/Microsoft.SqlServer.Server/InvalidUdtException.xml:30]:
```suggestion
This method Looks up a localized string similar to "&apos;{0}&apos; is an invalid user defined type, reason: {1}.", and fills it in order with the corresponding `udtType` and `resourceReason` values.  
```

File: doc/snippets/Microsoft.SqlServer.Server/InvalidUdtException.xml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            <summary>Creates a new <see cref="T:Microsoft.SqlServer.Server.InvalidUdtException" /> object.</summary>
+            <returns>A new <see cref="T:Microsoft.SqlServer.Server.InvalidUdtException" /> object.</returns>
+            <remarks>
+            <format type="text/markdown"><![CDATA[  
+  
+## Remarks  
+This method Looks up a localized string similar to "&apos;{0}&apos; is an invalid user defined type, reason: {1}.", and fills it in order with `udtType` and `resourceReason` resolved value.  
+
+> [!IMPORTANT]
+> This function exposes for backward compatibility, and should be used with default value for `resourceReason` parameter.
```
Comment (David-Engel) [inline doc/snippets/Microsoft.SqlServer.Server/InvalidUdtException.xml:33]:
```suggestion
> This function is exposed for backward compatibility and should be used with the default value for the `resourceReason` parameter.
```

File: doc/snippets/Microsoft.SqlServer.Server/SqlUserDefinedTypeAttribute.xml [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+                <see langword="true" /> if the user-defined type is byte ordered; otherwise <see langword="false" /> .</value>
+            <remarks>
+                <format type="text/markdown"><![CDATA[  
+  
+## Remarks  
+When set to `true`, the <xref:Microsoft.SqlServer.Server.SqlUserDefinedTypeAttribute.IsByteOrdered%2A> property in effect guarantees that the serialized binary data can be used for semantic ordering of the information. Thus, each instance of a byte-ordered UDT object can only have one serialized representation. When a comparison operation is performed in SQL Server on the serialized bytes, its results should be the same as if the same comparison operation had taken place in managed code.  
+
+The following features are supported when <xref:Microsoft.SqlServer.Server.SqlUserDefinedTypeAttribute.IsByteOrdered%2A> is set to `true`:  
+
+-   The ability to create indexes on columns of this type.
```
Comment (David-Engel) [inline doc/snippets/Microsoft.SqlServer.Server/SqlUserDefinedTypeAttribute.xml:53]:
```suggestion
- The ability to create indexes on columns of this type.  
```

File: doc/snippets/Microsoft.SqlServer.Server/SqlUserDefinedTypeAttribute.xml [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+                <format type="text/markdown"><![CDATA[  
+  
+## Remarks  
+When set to `true`, the <xref:Microsoft.SqlServer.Server.SqlUserDefinedTypeAttribute.IsByteOrdered%2A> property in effect guarantees that the serialized binary data can be used for semantic ordering of the information. Thus, each instance of a byte-ordered UDT object can only have one serialized representation. When a comparison operation is performed in SQL Server on the serialized bytes, its results should be the same as if the same comparison operation had taken place in managed code.  
+
+The following features are supported when <xref:Microsoft.SqlServer.Server.SqlUserDefinedTypeAttribute.IsByteOrdered%2A> is set to `true`:  
+
+-   The ability to create indexes on columns of this type.  
+
+-   The ability to create primary and foreign keys as well as CHECK and UNIQUE constraints on columns of this type.
```
Comment (David-Engel) [inline doc/snippets/Microsoft.SqlServer.Server/SqlUserDefinedTypeAttribute.xml:55]:
```suggestion
- The ability to create primary and foreign keys as well as CHECK and UNIQUE constraints on columns of this type.  
```

File: doc/snippets/Microsoft.SqlServer.Server/SqlUserDefinedTypeAttribute.xml [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+## Remarks  
+When set to `true`, the <xref:Microsoft.SqlServer.Server.SqlUserDefinedTypeAttribute.IsByteOrdered%2A> property in effect guarantees that the serialized binary data can be used for semantic ordering of the information. Thus, each instance of a byte-ordered UDT object can only have one serialized representation. When a comparison operation is performed in SQL Server on the serialized bytes, its results should be the same as if the same comparison operation had taken place in managed code.  
+
+The following features are supported when <xref:Microsoft.SqlServer.Server.SqlUserDefinedTypeAttribute.IsByteOrdered%2A> is set to `true`:  
+
+-   The ability to create indexes on columns of this type.  
+
+-   The ability to create primary and foreign keys as well as CHECK and UNIQUE constraints on columns of this type.  
+
+-   The ability to use Transact-SQL ORDER BY, GROUP BY, and PARTITION BY clauses. In these cases, the binary representation of the type is used to determine the order.
```
Comment (David-Engel) [inline doc/snippets/Microsoft.SqlServer.Server/SqlUserDefinedTypeAttribute.xml:57]:
```suggestion
- The ability to use Transact-SQL ORDER BY, GROUP BY, and PARTITION BY clauses. In these cases, the binary representation of the type is used to determine the order.  
```

File: doc/snippets/Microsoft.SqlServer.Server/SqlUserDefinedTypeAttribute.xml [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+The following features are supported when <xref:Microsoft.SqlServer.Server.SqlUserDefinedTypeAttribute.IsByteOrdered%2A> is set to `true`:  
+
+-   The ability to create indexes on columns of this type.  
+
+-   The ability to create primary and foreign keys as well as CHECK and UNIQUE constraints on columns of this type.  
+
+-   The ability to use Transact-SQL ORDER BY, GROUP BY, and PARTITION BY clauses. In these cases, the binary representation of the type is used to determine the order.  
+
+-   The ability to use comparison operators in Transact-SQL statements.
```
Comment (David-Engel) [inline doc/snippets/Microsoft.SqlServer.Server/SqlUserDefinedTypeAttribute.xml:59]:
```suggestion
- The ability to use comparison operators in Transact-SQL statements.  
```

File: doc/snippets/Microsoft.SqlServer.Server/SqlUserDefinedTypeAttribute.xml [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+
+-   The ability to create indexes on columns of this type.  
+
+-   The ability to create primary and foreign keys as well as CHECK and UNIQUE constraints on columns of this type.  
+
+-   The ability to use Transact-SQL ORDER BY, GROUP BY, and PARTITION BY clauses. In these cases, the binary representation of the type is used to determine the order.  
+
+-   The ability to use comparison operators in Transact-SQL statements.  
+
+-   The ability to persist computed columns of this type.
```
Comment (David-Engel) [inline doc/snippets/Microsoft.SqlServer.Server/SqlUserDefinedTypeAttribute.xml:61]:
```suggestion
- The ability to persist computed columns of this type.  
```

File: doc/snippets/Microsoft.SqlServer.Server/SqlUserDefinedTypeAttribute.xml [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+
+-   Not equal to (!=)  
+
+-   Greater than (>)  
+
+-   Less than (\<)  
+
+-   Greater than or equal to (>=)  
+
+-   Less than or equal to (<=)
```
Comment (David-Engel) [inline doc/snippets/Microsoft.SqlServer.Server/SqlUserDefinedTypeAttribute.xml:75]:
```suggestion
- Equal to (=)  

- Not equal to (!=)  

- Greater than (>)  

- Less than (\<)  

- Greater than or equal to (>=)  

- Less than or equal to (<=)  
```

File: doc/snippets/Microsoft.SqlServer.Server/SqlFunctionAttribute.xml [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+        <Name>
+            <summary>The name under which the function should be registered in SQL Server.</summary>
+            <value>A <see cref="T:System.String" /> value representing the name under which the function should be registered.</value>
+            <remarks>
+                <format type="text/markdown"><![CDATA[  
+
+## Remarks  
+This attribute is used only by Microsoft Visual Studio to automatically register the specified method as a user-defined function. It is not used by SQL Server.  
+
+Thee following example specifies that the user-defined function is referenced using the name `sp_scalarFunc`.
```
Comment (David-Engel) [inline doc/snippets/Microsoft.SqlServer.Server/SqlFunctionAttribute.xml:88]:
```suggestion
The following example specifies that the user-defined function is referenced using the name `sp_scalarFunc`.  
```

**Review by David-Engel — APPROVED** [review]

**Review by johnnypham — APPROVED** [review]

**Review by Kaur-Parminder — APPROVED** [review]

**Pull request merged**